passage/src/main/java/id/passage/android/PassageMagicLink.kt

```diff
@@ -0,0 +1,148 @@
+                identifier = identifier,
+                language = Passage.language,
+                magicLinkPath = magicLinkPath,
+            )
+        val response =
+            try {
+                loginAPI.loginMagicLink(Passage.appId, request)
+            } catch (e: Exception) {
+                throw MagicLinkLoginException.convert(e)
+            }
+        return response.magicLink
+    }
+
+    /**
+     * Activate Magic Link
+     *
+     * Authenticate a magic link for a user. This endpoint checks that the magic link is valid,
+     * then returns an authentication token for the user.
+     * @param userMagicLink full magic link that starts with "ml" (sent via email or text to the user)
+     * @return PassageAuthResult?
+     * @throws MagicLinkActivateException
+     */
+    suspend fun activate(userMagicLink: String): AuthResult {
+        val magicLinkAPI = MagicLinkAPI(Passage.BASE_PATH, passageClient)
+        val request = ActivateMagicLinkRequest(userMagicLink)
+        val response =
+            try {
+                magicLinkAPI.activateMagicLink(Passage.appId, request)
+            } catch (e: Exception) {
+                throw MagicLinkActivateException.convert(e)
+            }
+        // TODO: Once BE issue is fixed, we won't need to transform data model
+        val authResult =
+            AuthResult(
+                authToken = response.authResult.authToken,
+                redirectUrl = response.authResult.redirectUrl,
+                refreshToken = response.authResult.refreshToken,
+                refreshTokenExpiration = response.authResult.refreshTokenExpiration,
+            )
+        tokenStore.setTokens(authResult)
+        return authResult
+    }
+
+    /**
+     * Magic Link Status
+     *
+     * Check if a magic link has been activated yet or not. Once the magic link has been activated,
+     * this endpoint will return an authentication token for the user. This endpoint can be used to
+     * initiate a login in one device and then poll and wait for the login to complete on another
+     * device.
+     * @param magicLinkId Magic Link ID
+     * @return PassageAuthResult?
+     * @throws GetMagicLinkStatusException
+     */
+    suspend fun status(magicLinkId: String): AuthResult? {
+        val magicLinkAPI = MagicLinkAPI(Passage.BASE_PATH, passageClient)
+        val request = GetMagicLinkStatusRequest(magicLinkId)
+        val response =
+            try {
+                magicLinkAPI.magicLinkStatus(Passage.appId, request)
+            } catch (e: Exception) {
+                val exception = GetMagicLinkStatusException.convert(e)
+                if (exception is GetMagicLinkStatusNotFoundException) {
+                    Log.w(Passage.TAG, "Magic link not activated.")
+                    return null
+                }
+                throw exception
+            }
+        // TODO: Once BE issue is fixed, we won't need to transform data model
```

Comment:
Same comment as in `activate` method, hopefully this is fixed?

Reply:
Yes, it's fixed. thanks for pointing this out